app/src/main/java/chat/rocket/android/activity/LoginContract.java

```diff
@@ -0,0 +1,17 @@
+package chat.rocket.android.activity;
+
+import chat.rocket.android.shared.BaseContract;
+
+public interface LoginContract {
+
+  interface View extends BaseContract.View {
+    void showLogin(String hostname);
+
+    void showRetryLogin(String hostname);
+
+    void close();
```

Comment:
`close()` is not good. 


For example: 

```
    if (hostname == null || hostname.length() == 0) {
      view.close();
      return;
    }
```

It is hard to guess what `view.close()` does.

Please use another name like `hideLogin()`